app/services/invoices/calculate_fees_service.rb

```diff
@@ -108,11 +108,19 @@ def create_charges_fees(subscription, boundaries)
       subscription
         .plan
         .charges
+        .includes(:billable_metric)
         .joins(:billable_metric)
         .where(invoiceable: true)
         .where
         .not(pay_in_advance: true, billable_metric: { recurring: false })
         .each do |charge|
+        if charge.pay_in_advance? &&
```

Comment:
What about moving all this conditions in a method, for example `should_create_charge_fee?(charge)`?

Reply:
Maybe also, you could add a small spec for this one. What do you think?

Reply:
added:)